### PR TITLE
feat: Upgrade to .NET 8 Preview 7

### LINKED ISSUE #13183: Add support for .NET 8 Preview 7
## Why is this needed:

WPF Open Folder Dialog

## For which Platform:

All

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #13183

## PR Type

What kind of change does this PR introduce?

- Feature

### <samp>🤖 Generated by Copilot at e3f27f8</samp>

This pull request updates the .NET 8 SDK and tools version and adds a using directive for the `InputHelper` class in the integration tests. These changes are part of the effort to support .NET 8 in Uno Platform.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -69,9 +69,18 @@
...
-	</PropertyGroup>
+	<Choose>
+		<When Condition="$(TargetFramework.StartsWith('net7.0'))">
+			<PropertyGroup>
+				<MicrosoftNetCompilersToolsetVersion>4.6.0</MicrosoftNetCompilersToolsetVersion>
+			</PropertyGroup>
+		</When>
+		<When Condition="$(TargetFramework.StartsWith('net8.0'))">
+			<PropertyGroup>
+				<MicrosoftNetCompilersToolsetVersion>4.7.0-2.final</MicrosoftNetCompilersToolsetVersion>
```
Comment (Youssef1313) [inline src/Directory.Build.props:80]: Let's use the same compiler version in both cases. It is safe to upgrade to 4.7.0-2.final without conditions.

File: src/Uno.UI.RuntimeTests/IntegrationTests/common/TestServices.InputHelper.cs [hunk 2]
```diff
@@ -9,6 +9,7 @@
 using Uno.UI.RuntimeTests.Tests.Uno_UI_Xaml_Core;
 using System.Threading;
 using Windows.UI.Xaml.Media;
+using Uno.UI.RuntimeTests.Helpers;
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/IntegrationTests/common/TestServices.InputHelper.cs:12]: I think this change is not needed

Comment (Youssef1313) [reply src/Uno.UI.RuntimeTests/IntegrationTests/common/TestServices.InputHelper.cs:12]: Hmm.. What is the error?

**Review by Youssef1313 — APPROVED** [review]
LGTM, with a small comment/question about the change in `TestServices.InputHelper.cs`

File: src/Uno.UI.RuntimeTests/IntegrationTests/common/TestServices.InputHelper.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UI.RuntimeTests/IntegrationTests/common/TestServices.InputHelper.cs:12]: Ah, seems it is missing the using on master too https://github.com/unoplatform/uno/runs/15750427713 . So somehow we merged it without the using 😅

**Comment by Youssef1313** [comment]
I have been looking into the linker failure here for a while.

The failure is that `NullableContextAttribute` and `NullableAttribute` can't be resolved.

I think the problem is somehow with the runtime bits shipped by the Wasm bootstrapper.

If we compare these dlls:

- `C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App\8.0.0-preview.7.23375.6\System.Runtime.dll`
- `C:\Users\PC\AppData\Local\Temp\dotnet-runtime-wasm-windows-7224aa4-57cfd062405-5385526085-Release\runtimes\browser-wasm\lib\net8.0\System.Runtime.dll` (path can differ, but that is produced by the bootstrapper)

The former has a `TypeForwardedTo` for both attributes to forward it to `System.Private.CoreLib.dll`, while the latter doesn't.

**Comment by Youssef1313** [comment]
It could be https://github.com/dotnet/runtime/issues/85612 that broke us.

**Review by Youssef1313 — APPROVED** [review]

**Pull request merged**